File: back_spring_boot/common/dog-service/src/main/java/com/example/dogservice/entities/Dog.java
package com.example.dogservice.entities;

import com.example.dogservice.enums.StateDog;
import lombok.*;

import javax.persistence.*;

@Table(name= "dog")
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Dog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String gender;
    private String name;
    private String color;
    @Column(length = 2000, columnDefinition = "TEXT")
    private String description;
    private double weight;
    @Column(columnDefinition = "varchar(25) default 'MALADE'")
    @Enumerated(EnumType.STRING)
    private StateDog state;
    @Column(columnDefinition = "double default 0")
    private double price;
    private String image;
    private int age;
    private String categoryID;
}

File: argon-dashboard-angular-master/src/app/common/dog.ts
export class Dog {

  constructor
  (
    public id: number,
    public color: String,
    public description: String,
    public gender: String,
    public price: number,
    public state: String,
    public weight: number,
    public name: String,
    public image: String,
    public age: number,
    public categoryID: String
  ) {
  }
}

interface category {
  nomCategory: String
}
